src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -5571,6 +5571,1125 @@ .maxstack  1
+    protected virtual delegate*<void> M2() => throw null;
+}
+unsafe class Derived : Base
+{
+    protected override void M1(delegate* cdecl<void> ptr) {}
+    protected override delegate* cdecl<void> M2() => throw null;
+}");
+
+            comp.VerifyDiagnostics(
+                // (9,29): error CS0115: 'Derived.M1(delegate*<void>)': no suitable method found to override
+                //     protected override void M1(delegate* cdecl<void> ptr) {}
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "M1").WithArguments("Derived.M1(delegate*<void>)").WithLocation(9, 29),
+                // (10,46): error CS0508: 'Derived.M2()': return type must be 'delegate*<void>' to match overridden member 'Base.M2()'
+                //     protected override delegate* cdecl<void> M2() => throw null;
+                Diagnostic(ErrorCode.ERR_CantChangeReturnTypeOnOverride, "M2").WithArguments("Derived.M2()", "Base.M2()", "delegate*<void>").WithLocation(10, 46)
+            );
+        }
+
+        [Theory]
+        [InlineData(" ", "ref ")]
+        [InlineData(" ", "out ")]
+        [InlineData(" ", "in ")]
+        [InlineData("ref ", "")]
+        [InlineData("ref ", "out ")]
+        [InlineData("ref ", "in ")]
+        [InlineData("out ", " ")]
+        [InlineData("out ", "ref ")]
+        [InlineData("out ", "in ")]
+        [InlineData("in ", " ")]
+        [InlineData("in ", "ref ")]
+        [InlineData("in ", "out ")]
+        public void Override_RefnessMustMatch_Parameters(string refKind1, string refKind2)
+        {
+            var comp = CreateCompilationWithFunctionPointers(@$"
+unsafe class Base
+{{
+    protected virtual void M1(delegate*<{refKind1}string, void> ptr) {{}}
+    protected virtual delegate*<{refKind1}string, void> M2() => throw null;
+}}
+unsafe class Derived : Base
+{{
+    protected override void M1(delegate*<{refKind2}string, void> ptr) {{}}
+    protected override delegate*<{refKind2}string, void> M2() => throw null;
+}}");
+
+            comp.VerifyDiagnostics(
+                // (9,29): error CS0115: 'Derived.M1(delegate*<{refKind2} string, void>)': no suitable method found to override
+                //     protected override void M1(delegate*<{refKind2} string, void> ptr) {}
+                Diagnostic(ErrorCode.ERR_OverrideNotExpected, "M1").WithArguments($"Derived.M1(delegate*<{refKindString(refKind2)}string, void>)").WithLocation(9, 29),
+                // (10,49): error CS0508: 'Derived.M2()': return type must be 'delegate*<{refKind1} string, void>' to match overridden member 'Base.M2()'
+                //     protected override delegate*<{refKind2} string, void> M2() => throw null;
+                Diagnostic(ErrorCode.ERR_CantChangeReturnTypeOnOverride, "M2").WithArguments("Derived.M2()", "Base.M2()", $"delegate*<{refKindString(refKind1)}string, void>").WithLocation(10, 48 + refKind2.Length)
+            );
+
+            static string refKindString(string orig) => orig == " " ? "" : orig;
```

Comment:
>refKindString [](start = 26, length = 13)

Could this local function be removed if the `RefKind.None` cases were "" rather than " "?